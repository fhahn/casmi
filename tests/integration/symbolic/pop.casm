// symbolic

CASM test

function (symbolic) a: -> List(Int) initially { [1, 2] }

rule main = {
  
  pop x from a

  program( self ) := undef
}

init main

