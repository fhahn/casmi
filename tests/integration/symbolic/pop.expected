forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,fcons(eEmptyList,1,sym2)).
tff(symbolNext, type, sym3: $int).
fof(id1,hypothesis,fcons(sym2,2,sym3)).
fof(id2,hypothesis,sta(1,sym3)).%CREATE: a
tff(symbolNext, type, sym4: $int).
fof(id3,hypothesis,fpop(sym3, 1, sym4)).
fof(id4,hypothesis,sta(2,sym4)).%UPDATE: a
fof(final0,hypothesis,sta(0,sym4)).%FINAL: a

